storage/innobase/os/os0file.cc

```diff
@@ -3387,35 +3387,50 @@ int os_aio_init() noexcept
   int max_read_events= int(srv_n_read_io_threads *
                            OS_AIO_N_PENDING_IOS_PER_THREAD);
   int max_events= max_read_events + max_write_events;
-  int ret;
-#if LINUX_NATIVE_AIO
-  if (srv_use_native_aio && !is_linux_native_aio_supported())
-    goto disable;
-#endif
-
-  ret= srv_thread_pool->configure_aio(srv_use_native_aio, max_events);
+  int ret= 1;
 
-#ifdef LINUX_NATIVE_AIO
-  if (ret)
+#ifdef __linux__
+  if (srv_use_native_aio)
   {
-    ut_ad(srv_use_native_aio);
-disable:
-    ib::warn() << "Linux Native AIO disabled.";
-    srv_use_native_aio= false;
-    ret= srv_thread_pool->configure_aio(false, max_events);
-  }
+    switch (srv_linux_aio_method)
```

Comment:
Suggestion : I think most of this `if linux` fallback logic (uring to libaio) could be handled inside tpool::configure_aio itself, you got the OS_DEFAULT for it, right? Will be less clutter here and you can put OS_DEFAULT  to use

Reply:
`sql_print_warning()` is not being used in `tpool`. What we can do is to remove some `#ifdef LINUX_NATIVE_AIO` checks that I believe are redundant by now.